src/runtime/contrib/json/json_node.h

```diff
@@ -0,0 +1,358 @@
+    }
+  }
+
+  /*!
+   * \brief Check if a node is a leaf node, i.e. input to the graph.
+   *
+   * \return True if the node has no input, otherwise, false.
+   */
+  bool IsLeaf() const { return inputs_.empty(); }
+
+  /*!
+   * \brief Return the number of outputs of the node.
+   *
+   * \return The number of the output.
+   */
+  uint32_t GetNumOutput() const { return num_outputs_; }
+
+  /*!
+   * \brief Return the input entries.
+   *
+   * \return The input entries.
+   */
+  std::vector<JSONGraphNodeEntry> GetInputs() const { return inputs_; }
+
+  /*!
+   * \brief Return the op type.
+   *
+   * \return The op type.
+   */
+  std::string GetOpType() const { return op_type_; }
+
+  /*!
+   * \brief Return the op name.
+   *
+   * \return The op name.
+   */
+  std::string GetOpName() const { return name_; }
+
+  /*!
+   * \brief Return the op output shapes.
+   *
+   * \return The shapes.
+   */
+  std::vector<std::vector<int64_t>> GetOpShape() const { return shape_; }
+
+  /*!
+   * \brief Return the op types.
+   *
+   * \return The types.
+   */
+  std::vector<DLDataType> GetOpDataType() const { return dtype_; }
+
+  /*!
+   * \brief Set the number of outputs of the node.
+   *
+   * \param num_outputs The number of output.
+   */
+  void SetNumOutput(uint32_t num_outputs) { num_outputs_ = num_outputs; }
+
+  /*!
+   * \brief Get the value of an attribute in the node.
+   *
+   * \tparam T The return type.
+   * \param key The key for lookup.
+   *
+   * \return The value.
+   */
+  template <typename T>
+  T GetAttr(const std::string& key) const {
+    CHECK_GT(attrs_.count(key), 0U) << "Key: " << key << "is not found";
+    return dmlc::get<T>(attrs_.at(key));
+  }
+
+  /*!
+   * \brief Set an attribute for the node.
+   *
+   * \tparam ValueT The type of the value being stored.
+   * \param key The key of the attribute.
+   * \param value The value of the attribute.
+   */
+  template <typename ValueT>
+  void SetAttr(const std::string& key, const ValueT& value) {
+    attrs_[key] = value;
+  }
+
+  virtual ~JSONGraphNode() {}
+
+ private:
+  /*! \brief The number of input. */
+  uint32_t num_inputs_{0};
+  /*! \brief The number of output. */
+  uint32_t num_outputs_{1};
+  /*! \brief The name of the op. It is the symbol that used for runtime lookup. */
+  std::string name_;
+  /*! \brief The operator type, i.e. input is "null". */
+  std::string op_type_;
+  /*! \brief The shape of the node. */
+  std::vector<std::vector<int64_t>> shape_;
+  /*! \brief The type of the node. */
+  std::vector<DLDataType> dtype_;
+  /*! \brief The inputs of the node. */
+  std::vector<JSONGraphNodeEntry> inputs_;
+  /*!
+   * \brief Attribute of the node. For simplicity, we store all attribute as
+   * a list of std::string. It's the developer's resposibility to check the
+   * required attribute of a certain op and convert it into the needed type.
+   *
+   * For example, for conv2d, this map could contain:
+   *  attrs_["strides"] = ["1", "1"]
```

Comment:
Is this what you are looking for or you are referring to somethig else?
https://github.com/apache/incubator-tvm/pull/5919/files/611411df6ec868a34295a240e080c3339b760de1#diff-e7204ab48c8a7f5b4acbd341838634e3R323

Reply:
I also hit this error

```
Traceback (most recent call last):
  File "test_pass_partition_graph.py", line 1294, in <module>
    test_extern_dnnl()
  File "test_pass_partition_graph.py", line 455, in test_extern_dnnl
    (1, 32, 14, 14), ref_res.asnumpy(), tol=1e-5)
  File "test_pass_partition_graph.py", line 231, in check_result
    check_vm_result()
  File "test_pass_partition_graph.py", line 199, in check_vm_result
    exe = relay.vm.compile(mod, target=target, params=params)
  File "/home/masa/projects/dev/tvm/python/tvm/relay/backend/vm.py", line 70, in compile
    compiler.codegen()
  File "/home/masa/projects/dev/tvm/python/tvm/relay/backend/vm.py", line 139, in codegen
    self._codegen()
  File "/home/masa/projects/dev/tvm/python/tvm/_ffi/_ctypes/packed_func.py", line 225, in __call__
    raise get_last_ffi_error()
tvm._ffi.base.TVMError: Traceback (most recent call last):
  [bt] (8) /home/masa/projects/dev/tvm/build/libtvm.so(TVMFuncCall+0x5f) [0x7f211cc4bbcf]
  [bt] (7) /home/masa/projects/dev/tvm/build/libtvm.so(+0xda8ae9) [0x7f211cb11ae9]
  [bt] (6) /home/masa/projects/dev/tvm/build/libtvm.so(tvm::relay::vm::VMCompiler::Codegen()+0x628) [0x7f211cb10fe8]
  [bt] (5) /home/masa/projects/dev/tvm/build/libtvm.so(tvm::relay::CompileEngineImpl::LowerExternalFunctions()+0x6f7) [0x7f211cac10c7]
  [bt] (4) /home/masa/projects/dev/tvm/build/libtvm.so(tvm::runtime::TypedPackedFunc<tvm::runtime::Module (tvm::runtime::ObjectRef const&)>::AssignTypedLambda<tvm::runtime::Module (*)(tvm::runtime::ObjectRef const&)>(tvm::runtime::Module (*)(tvm::runtime::ObjectRef const&))::{lambda(tvm::runtime::TVMArgs const&, tvm::runtime::TVMRetValue*)#1}::operator()(tvm::runtime::TVMArgs const&, tvm::runtime::TVMRetValue*) const+0xf8) [0x7f211cc25238]
  [bt] (3) /home/masa/projects/dev/tvm/build/libtvm.so(tvm::relay::contrib::DNNLCompiler(tvm::runtime::ObjectRef const&)+0xa52) [0x7f211cc2c572]
  [bt] (2) /home/masa/projects/dev/tvm/build/libtvm.so(tvm::relay::backend::contrib::JSONSerializer::Save(dmlc::JSONWriter*)+0x58e) [0x7f211cc32eee]
  [bt] (1) /home/masa/projects/dev/tvm/build/libtvm.so(tvm::runtime::json::JSONGraphNode::Save(dmlc::JSONWriter*)+0x9f3) [0x7f211cc32153]
  [bt] (0) /home/masa/projects/dev/tvm/build/libtvm.so(+0xec2938) [0x7f211cc2b938]
  File "/home/masa/projects/dev/tvm/3rdparty/dmlc-core/include/dmlc/././json.h", line 590
TVMError: Check failed: it != nmap.end() && it->first == id: Type St6vectorIN4dmlc3anyESaIS1_EE has not been registered via DMLC_JSON_ENABLE_ANY
```

Reply:
Hmm...we hit this before and that's why we put the above changes. Let me double check

Reply:
The seems to be about `std::vector`, so I don't think `Handler<std::unordered_map<std::string, dmlc::any>>` is related?

Reply:
It is not about current handler, it is about the specialization of struct Handler<std::vector<dmlc::any>>

Reply:
Can't reproduce the issue on my local. I'll use another instance to try again. Meanwhile, could you check if you have run `git submodule update` to update the dmlc dependency?

Reply:
I launched a new instance to run the unit tests but still didn't encounter this issue. Maybe you can clone a new one and try again? We will also work on enabling DNNL on the CI so that we can see if this is an environment issue.

Reply:
ok my local copy was pointing to the commit 2 days ago. Updating to the latest fixed this. 

Somehow the change that added `Handler<std::unordered_map<std::string, dmlc::any>>` did the trick?

Reply:
yeah

Reply:
Thanks all for looking into this, using the latest commit worked